src/vm/excep.cpp

```diff
@@ -7556,7 +7556,7 @@ void InitSavedExceptionInfo()
 void FaultingExceptionFrame::Init(CONTEXT *pContext)
 {
     WRAPPER_NO_CONTRACT;
-#if defined(_TARGET_X86_)
+#if defined(_TARGET_X86_) && !defined(WIN64EXCEPTIONS)
```

Comment:
It seems that in many places where you use this `#if` condition, it would make sense to use `#ifndef (WIN64EXCEPTIONS)` with `#ifdef _TARGET_X86_ ... #else PORTABILITY_ASSERT #endif` in it.